src/ClaseServiciu/ServiciuCitire.java

```diff
@@ -0,0 +1,103 @@
+package ClaseServiciu;
+
+import Clase.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class ServiciuCitire {
+    public Angajat[] citireAngajati() throws IOException{
+        Angajat[] ang = {};
+        try {
+            String path = System.getProperty("user.dir") + "\\src\\FisiereCSV\\Angajat.txt";
+            Scanner scaner = new Scanner(new File(path));
+            scaner.useDelimiter(",");
+            Bibliotecar a1 = new Bibliotecar(scaner.next(),scaner.next(),scaner.next());
+            Bodyguard a2 = new Bodyguard(scaner.next().strip(),scaner.next(),scaner.next());
+
+            ang = Arrays.copyOf(ang, ang.length + 1);
+            ang[ang.length - 1] = a1;
+            ang = Arrays.copyOf(ang, ang.length + 1);
+            ang[ang.length - 1] = a2;
+
+            return ang;
+        } catch (IOException e) {
+            System.out.print("eroare la citire");
+        }
+        return ang;
+    }
+
+    public Autor[] citireAutori() throws IOException{
+        Autor[] aut = {};
+        try {
+            String path = System.getProperty("user.dir") + "\\src\\FisiereCSV\\Autor.txt";
+            Scanner scaner = new Scanner(new File(path));
+            scaner.useDelimiter(",");
+            while (scaner.hasNext()) {
+                Autor a = new Autor(scaner.next().strip(),scaner.next(),scaner.next(),scaner.next());
+                aut = Arrays.copyOf(aut, aut.length + 1);
+                aut[aut.length - 1] = a;
+            }
+            return aut;
+        } catch (IOException e) {
+            System.out.print("eroare la citire");
+        }
+        return aut;
+    }
+
+    public Carte[] citireCarti() throws IOException{
+        Carte[] c = {};
+        try {
+            String path = System.getProperty("user.dir") + "\\src\\FisiereCSV\\Carte.txt";
+            Scanner scaner = new Scanner(new File(path));
+            scaner.useDelimiter(",");
+            while (scaner.hasNext()) {
+                String numeCarte = scaner.next().strip();
+                Autor a = new Autor(scaner.next(),scaner.next(),scaner.next(),scaner.next());
+                Carte crt = new Carte(numeCarte, a);
+                c = Arrays.copyOf(c, c.length + 1);
+                c[c.length - 1] = crt;
+            }
+            return c;
+        } catch (IOException e) {
+            System.out.print("eroare la citire");
```

Comment:
maybe you can use different exception messages at reading part for author, employee